File: lua/tests/speedtest.lua
--[[
Copyright 2010 Google Inc.
All Rights Reserved.

Diff Speed Test

fraser@google.com
--]]

package.path = package.path .. ';../?.lua'
local dmp = require 'diff_match_patch'

function main()
  text1 = readlines('speedtest1.txt')
  text2 = readlines('speedtest2.txt')

  dmp.settings{ Diff_Timeout=0 }

  -- Execute one reverse diff as a warmup.
  dmp.diff_main(text2, text1, false)
  collectgarbage('collect')

  start_time = os.clock()
  dmp.diff_main(text1, text2, false)
  end_time = os.clock()
  print('Elapsed time: ' .. (end_time - start_time))
end

function readlines(filename)
  f = io.open(filename, 'rb')
  text = f:read('*a')
  f:close()
  return text
end

main()

File: csharp/tests/Speedtest.cs
// Copyright 2010 Google Inc.
// All Right Reserved.

/*
 * To compile with Mono:
 *   mcs Speedtest.cs ../DiffMatchPatch.cs
 * To run with Mono:
 *   mono Speedtest.exe
*/

using DiffMatchPatch;
using System;
using System.Collections.Generic;

public class Speedtest {
  public static void Main(string[] args) {
    string text1 = System.IO.File.ReadAllText("Speedtest1.txt");
    string text2 = System.IO.File.ReadAllText("Speedtest2.txt");

    diff_match_patch dmp = new diff_match_patch();
    dmp.Diff_Timeout = 0;

    // Execute one reverse diff as a warmup.
    dmp.diff_main(text2, text1);
    GC.Collect();
    GC.WaitForPendingFinalizers();

    DateTime ms_start = DateTime.Now;
    dmp.diff_main(text1, text2);
    DateTime ms_end = DateTime.Now;

    Console.WriteLine("Elapsed time: " + (ms_end - ms_start));
  }
}
